### PR TITLE
feat(compiler)!: Add additional reserved keywords

### PR DESCRIPTION
**Pull request by spotandjake**
Added `module`, `macro`, `yield` to Reserved Keywords List, with the goal of making future breaking changes less intrusive. 

If there are any additional reserved keywords that would make sense to add let me know.

Closes: #1463

### CODE REVIEW

**Review by ospencer — APPROVED** [review]

**Review by phated — CHANGES_REQUESTED** [review]
I'm not sold on these keywords so I don't really want to reserve them.

File: compiler/src/parsing/lexer.re [hunk 1]
```diff
@@ -249,6 +249,11 @@ let rec token = lexbuf => {
   | "try" => positioned(TRY)
   | "throw" => positioned(THROW)
   | "catch" => positioned(CATCH)
+  | "async" => positioned(ASYNC)
+  | "await" => positioned(AWAIT)
```
Comment (ospencer) [inline compiler/src/parsing/lexer.re:253]: @phated doesn't want to reserve these two, so let's remove them.

**Review by ospencer — CHANGES_REQUESTED** [review]

**Comment by spotandjake** [comment]
Made those changes

**Review by ospencer — APPROVED** [review]

File: compiler/src/parsing/lexer.re (hunk 1 above)
Comment (HTG-YT) [reply compiler/src/parsing/lexer.re]: Is there explicit reason why though? Perhaps it may come in handy in the future-

Comment (spotandjake) [reply compiler/src/parsing/lexer.re]: Don't want to lock us into a system we might not use.

**Review by marcusroberts — APPROVED** [review]

File: compiler/src/parsing/lexer.re [hunk 2]
```diff
@@ -250,6 +250,9 @@ let rec token = lexbuf => {
   | "try" => positioned(TRY)
   | "throw" => positioned(THROW)
   | "catch" => positioned(CATCH)
+  | "module" => positioned(MODULE)
```
Comment (ospencer) [inline compiler/src/parsing/lexer.re:253]: Just want to preemptively avoid a merge conflict here :P 

```suggestion
```

File: compiler/src/parsing/parser.mly [hunk 3]
```diff
@@ -42,7 +42,7 @@ module Grain_parsing = struct end
 %token EOL EOF
 
 // reserved tokens
-%token TRY CATCH COLONCOLON
+%token TRY CATCH COLONCOLON MODULE MACRO YIELD
```
Comment (ospencer) [inline compiler/src/parsing/parser.mly:45]:
```suggestion
%token TRY CATCH COLONCOLON MACRO YIELD
```

**Review by phated — APPROVED** [review]

**Pull request merged**